### DISCUSSION TITLE
is it possible to add unstructured?

### DISCUSSION THREAD

**Discussion by legaltextai** [infiniflow/ragflow#1406] [General]
for the parsing step, in my hosted version. 
thank you.

**Comment by KevinHuSh** [comment]
Could you make an example about the usage?

**Reply by legaltextai** [reply]
do you mean use case or actual code? 
i 've only played with the cloud version of ragflow, which default parser do you use for both pdfs and png?

**Reply by KevinHuSh** [reply]
Use case or code, both are fine
PDF parser has a lot of models: layout recognize, TSR.
PNG is using OCR or multi-module LLM to gain text description for the picture.

**Comment by legaltextai** [comment]
which pdf parser do you use now in your default settings? 
unstructured is the best library for complicated documents for self-hosted parsing

**Reply by KevinHuSh** [reply]
It's developed by our own.

**Comment by legaltextai** [comment]
is the parser run locally or via API if I install on my host?

**Reply by KevinHuSh** [reply]
Locally.
